### PR TITLE
feat: Add tag entity

### PR DESCRIPTION
**Pull request by 0sinf**
## What does this PR do?

Add tag entity that had relationship with bookmark. (many-to-many).

## Type of change

- [x] New feature
- [ ] Bug fix
- [ ] Code improvements
- [ ] Breaking change
- [ ] Documentation
- [ ] etc.

## Changes

- Add tag entity
- Add many-to-many relation. (bookmark, tag)
- When create bookmark, find or save tag.
- Fix response that find a group. (Add tags in bookmark)

### CODE REVIEW

File: src/group/dto/findone-group.dto.ts [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
-
 export interface FindOneRes {
-  bookmarks: BookmarkEntity[];
+  bookmarks: {
+    title: string;
+    description: string;
+    link: string;
+    image: string;
+    tags: string[];
+  }[];
```
Comment (marshallku) [inline src/group/dto/findone-group.dto.ts]: How about making bookmark instead?

File: src/group/group.service.ts [hunk 2]
```diff
@@ -96,7 +97,13 @@ export class GroupService {
       hasNext,
     };
 
-    return { bookmarks, pagination };
+    return {
+      bookmarks: bookmarks.map((bookmark) => ({
+        ...bookmark,
+        tags: bookmark.tags.map((tag) => tag.tag),
```
Comment (marshallku) [inline src/group/group.service.ts]: Destructuring `tag` to `({ tag })` might looks better.

Comment (0sinf) [reply src/group/group.service.ts]: Good Idea! 👏🏻

File: src/group/dto/findone-group.dto.ts (hunk 1 above)
Comment (0sinf) [reply src/group/dto/findone-group.dto.ts]: Good Idea! 👏🏻

**Review by marshallku — APPROVED** [review]
LGTM!

**Pull request merged**